data_structures/skip_list.cpp

```diff
@@ -0,0 +1,207 @@
+/**
+ * A skip list is a data structure that is used for storing a sorted list of items with a
+ *  help of hierarchy of linked lists that connect increasingly sparse subsequences of the items
+ * 
+ *  References used: GeeksForGeeks skip list code, https://iq.opengenus.org/skip-list/ Pseudo Code.
+*/
+
+#include <iostream>
+#include <vector>
+#include <string.h>
+#include <stdlib.h>
+#include <ctime>   
+
+// using namespace std; 
+using std::vector;
+using std::endl;
+
+#define MAXLVL 2   ///< maximum level of skip list
```

Comment:
use a better macro name
```suggestion
#define MAX_LEVEL 2   ///< maximum level of skip list
```
can this value be increased? if yes, to what limit? if not, why?

Reply:
Yes it can, depends on the problem. For most of the cases it's better be log(n) where n is number of elements but if I remember correctly Skip Lists are also used on sparse arrays where some everistics are need.